src/app/ipv/middleware.js

```diff
@@ -238,9 +238,9 @@ module.exports = {
         return res.render(`ipv/${req.session.currentPage}.njk`, {
           context: "unrecoverable",
         });
-      } else if (pageId === "pyi-timeout-unrecoverable") {
-        req.session.currentPage = "pyi-timeout-unrecoverable";
-        return res.render(`ipv/${req.session.currentPage}.njk`);
+      } else if (pageId === "pyi-timeout" && context === "unrecoverable") {
+        req.session.currentPage = "pyi-timeout";
+        return res.render(`ipv/${req.session.currentPage}.njk`, { context });
```

Comment:
Maybe I'm not reading this right, but why can't this just be handled in the switch below which already passes context to the template?

Reply:
This early escape route to the pyi-timeout-unrecoverable page has been there since I joined, so not sure on the original requirement to have this.

It looks like these early escapes before the switch statement are for handling forced redirects when there is a backend error, as opposed to the switch statement which handles the rendering from the usual journey-map routing

Reply:
I think removing this early-escape, would result in the pyi-timeout-unrecoverable redirect being caught in the else if statement below, resorting in the wrong page being rendered

Reply:
I see ok, thanks